### PR TITLE
refactor(eth sig): move chain id to message payload

### PR DESCRIPTION
**Pull request by Orland0x**

### CODE REVIEW

File: starknet/src/utils/eip712.cairo [hunk 1]
```diff
@@ -105,6 +101,7 @@ mod EIP712 {
         ) -> u256 {
             let encoded_data = array![
                 u256 { low: PROPOSE_TYPEHASH_LOW, high: PROPOSE_TYPEHASH_HIGH },
+                Felt252IntoU256::into(starknet::get_tx_info().unbox().chain_id),
```
Comment (pscott) [inline starknet/src/utils/eip712.cairo:104]:
```suggestion
                starknet::get_tx_info().unbox().chain_id.into(),
```

would that work?

File: tests/eth-sig-auth.test.ts [hunk 2]
```diff
@@ -90,6 +89,7 @@ describe('Ethereum Signature Authenticator', function () {
 
     // PROPOSE
     const proposeMsg: Propose = {
+      chainId: '0x534e5f474f45524c49',
```
Comment (pscott) [inline tests/eth-sig-auth.test.ts:92]: might be worth defining a constant `SN_GOERLI = ...` ?

**Review by pscott — APPROVED** [review]
utACK

File: starknet/src/utils/eip712.cairo (hunk 1 above)
Comment (Orland0x) [reply starknet/src/utils/eip712.cairo:104]: No because theres a namespace collision between the generic impl and the core lib one

Comment (pscott) [reply starknet/src/utils/eip712.cairo:104]: love it :sad-pepe:

**Pull request merged**